Setup Pages for a SaaS platform (#15179)
cloudflare/cloudflare-docs: content/pages/how-to/setup-pages-saas.md

---
pcx_content_type: how-to
title: Setup Pages for a SaaS platform
---

# Setup Pages for a SaaS platform

## Background

Learn how to have customers point their domains to your Pages projects for SaaS platforms. This allows customers to use Cloudflare proxying/DNS or an external DNS provider.

## Set up

{{<Aside type="note" header="SSL for SaaS">}}
  You should not use SSL for SaaS for this situation. Trying to use SSL for SaaS will hit problems due to Pages itself using SSL for SaaS underneath.
{{</Aside>}}

{{<Aside type="note" header="Domain limit">}}
  If you need the [custom domain limit](/pages/platform/limits/#custom-domains) raised and you're an Enterprise customer please contact your account team.
  The custom domain limit is _per project_ not per account.
{{</Aside>}}

> 1. Pick your CNAME target, the record that you want customers to point to. In this guide, our example target will be `custoemrs.example.com`.

Reviewer: ```suggestion
1. Pick your CNAME target, the record that you want customers to point to. In this guide, our example target will be `customers.example.com`.
```

2. Add your CNAME target to [Pages custom domains](/pages/configuration/custom-domains/)
  ![Add your CNAME target to Pages custom domains](/images/pages/how-to/setup-pages-saas/saas-domain.png)

1. With your CNAME target setup, you can now have customers add a CNAME on their own domains to `customers.example.com`. Example:
  ![Example customer record](/images/pages/how-to/setup-pages-saas/customer-cname.png)

1. If you're handling multiple customers within the same Pages project, you will probably want to serve different content per customer. To do this, within a [Pages Function](/pages/functions/) you can check the hostname of the request. Example:

> ```js

Reviewer: I think this js code needs to be indented

Author: Looks alright to me as is, couldn't get it to be indented

Reviewer: adding a three (+) spaces indentation seems the way to go for me
| Without indentation | With indentation |
|--------|--------|
|  ![Screenshot 2024-06-21 at 00 05 24](https://github.com/cloudflare/cloudflare-docs/assets/61631103/255ecda1-8681-4aeb-acf7-e380d609a5f2) |  ![Screenshot 2024-06-21 at 00 04 50](https://github.com/cloudflare/cloudflare-docs/assets/61631103/f6bffe38-127c-4f16-84a0-9b64d2bd09a4) |

---
pcx_content_type: how-to
title: Setup Pages for a SaaS platform
---

# Setup Pages for a SaaS platform

> ## Background

Reviewer 2: ```suggestion
```

Learn how to have customers point their domains to your Pages projects for SaaS platforms. This allows customers to use Cloudflare proxying/DNS or an external DNS provider.

## Set up

> {{<Aside type="note" header="SSL for SaaS">}}

Reviewer 2: I'd move both these asides to a new section called, `considerations`

>   You should not use SSL for SaaS for this situation. Trying to use SSL for SaaS will hit problems due to Pages itself using SSL for SaaS underneath.

Reviewer 2: ```suggestion
  You should not use [Cloudflare for SaaS](/cloudflare-for-platforms/cloudflare-for-saas/) for this situation. Pages is also using Cloudflare for SaaS and another layer of it would cause conflicts.
```
